src/include/btree_inline.h

```diff
@@ -2054,20 +2050,21 @@ __wt_btcur_skip_page(WT_CURSOR_BTREE *cbt)
      * We are making these decisions while holding a lock for the page as checkpoint or eviction can
      * make changes to the data structures (i.e., aggregate timestamps) we are reading.
      */
-    if (session->txn->isolation == WT_ISO_SNAPSHOT && !__wt_page_is_modified(page) &&
-      !F_ISSET(&cbt->iface, WT_CURSTD_IGNORE_TOMBSTONE) && previous_state == WT_REF_MEM) {
-
+    if (session->txn->isolation == WT_ISO_SNAPSHOT && !ignore_tombstone &&
+      previous_state == WT_REF_MEM) {
```

Comment:
This seems like a lot of work for something that can never be "true"? Why not pull the `session.txn.isolation` and `ignore_tombstone` checks to a higher level and don't specify a custom skip function if it's never going to be useful?

Reply:
If you pull the check for `WT_REF_MEM` to the set of `previous_state` and return immediately, you can outdent all of this one level and it will read more cleanly.

Reply:
I have pulled out the checks specific to the walk outside of the skip function and deciding to call the tree walk with or without custom skip based on the checks. I have also pulled check for the previous state together with setting the locked state, and return early in case we do not get the lock.

Looks much cleaner now.